File: src/main/java/tech/simter/jackson/javatime/JavaTimeModule.java
package tech.simter.jackson.javatime;

import com.fasterxml.jackson.databind.module.SimpleModule;

/**
 * @author RJ
 */
public class JavaTimeModule extends SimpleModule {
  public static final JavaTimeModule INSTANCE = new JavaTimeModule();

  public JavaTimeModule() {
    JavaTimeSerializer.addAllSupportedSerializerToModule(this);
    JavaTimeDeserializer.addAllSupportedDeserializerToModule(this);
    JavaTimeKeyDeserializer.addAllSupportedKeyDeserializerToModule(this);
  }
}

File: src/main/kotlin/tech/simter/jackson/javatime/JavaTimeModule.kt
package tech.simter.jackson.javatime

import com.fasterxml.jackson.databind.module.SimpleModule

/**
 * @author RJ
 */
class JavaTimeModule : SimpleModule() {
  init {
    JavaTimeSerializer.addAllSupportedSerializerToModule(this)
    JavaTimeDeserializer.addAllSupportedDeserializerToModule(this)
    JavaTimeKeyDeserializer.addAllSupportedKeyDeserializerToModule(this)
  }
}